backend/src/error.rs

```diff
@@ -0,0 +1,52 @@
+#![allow(clippy::module_name_repetitions)]
+use axum::response::{IntoResponse, Response};
+use color_eyre::eyre::Error;
+use hyper::StatusCode;
+use thiserror::Error;
+
+/// Generic error that wraps `eyre::Error`.
+///
+/// For the internal server errors
+pub struct AppError(Error);
```

Comment:
One big problem with `anyhow` and `eyre` is that it is much harder to check the error type, when special handling is needed. In general it is better to use `enum` based errors or some kind of combination of `enum` based error and `anyhow` error.

I googled and found this interesting crate: https://crates.io/crates/error-stack 
We can try to use it. Basically, it track the error stack with conversion and allow to access the top level error easily. Its combination with custom `enum` based error should do the thing